modules/openapi-generator/src/main/resources/powershell-experimental/rsa_provider.mustache

```diff
@@ -0,0 +1,377 @@
+
+        static byte[] GetOpenSSL3deskey(byte[] salt, SecureString secpswd, int count, int miter)
+        {
+            IntPtr unmanagedPswd = IntPtr.Zero;
+            int HASHLENGTH = 16;    //MD5 bytes
+            byte[] keymaterial = new byte[HASHLENGTH * miter];     //to store contatenated Mi hashed results
+
+
+            byte[] psbytes = new byte[secpswd.Length];
+            unmanagedPswd = Marshal.SecureStringToGlobalAllocAnsi(secpswd);
+            Marshal.Copy(unmanagedPswd, psbytes, 0, psbytes.Length);
+            Marshal.ZeroFreeGlobalAllocAnsi(unmanagedPswd);
+
+            //UTF8Encoding utf8 = new UTF8Encoding();
+            //byte[] psbytes = utf8.GetBytes(pswd);
+
+            // --- contatenate salt and pswd bytes into fixed data array ---
+            byte[] data00 = new byte[psbytes.Length + salt.Length];
+            Array.Copy(psbytes, data00, psbytes.Length);      //copy the pswd bytes
+            Array.Copy(salt, 0, data00, psbytes.Length, salt.Length); //concatenate the salt bytes
+
+            // ---- do multi-hashing and contatenate results  D1, D2 ...  into keymaterial bytes ----
+            MD5 md5 = new MD5CryptoServiceProvider();
+            byte[] result = null;
+            byte[] hashtarget = new byte[HASHLENGTH + data00.Length];   //fixed length initial hashtarget
+
+            for (int j = 0; j < miter; j++)
+            {
+                // ----  Now hash consecutively for count times ------
+                if (j == 0)
+                    result = data00;    //initialize 
+                else
+                {
+                    Array.Copy(result, hashtarget, result.Length);
+                    Array.Copy(data00, 0, hashtarget, result.Length, data00.Length);
+                    result = hashtarget;
+                    //Console.WriteLine("Updated new initial hash target:") ;
+                    //showBytes(result) ;
+                }
+
+                for (int i = 0; i < count; i++)
+                    result = md5.ComputeHash(result);
+                Array.Copy(result, 0, keymaterial, j * HASHLENGTH, result.Length);  //contatenate to keymaterial
+            }
+            //showBytes("Final key material", keymaterial);
+            byte[] deskey = new byte[24];
+            Array.Copy(keymaterial, deskey, deskey.Length);
+
+            Array.Clear(psbytes, 0, psbytes.Length);
+            Array.Clear(data00, 0, data00.Length);
+            Array.Clear(result, 0, result.Length);
+            Array.Clear(hashtarget, 0, hashtarget.Length);
+            Array.Clear(keymaterial, 0, keymaterial.Length);
+
+            return deskey;
+        }
+
+        public static string GetRSASignb64encode(string private_key_path, byte[] digest)
+        {
+            RSACryptoServiceProvider cipher = new RSACryptoServiceProvider();
+            cipher = GetRSAProviderFromPemFile(private_key_path);
+            RSAPKCS1SignatureFormatter RSAFormatter = new RSAPKCS1SignatureFormatter(cipher);
```

Comment:
Ah ok, finally I found the code location that configures the specific signature algorithm. This needs to be parameterized, with support for RSA PKCS v1.5 and RSA SSA. Also, this needs to be documentation in the user documentation.